### PR TITLE
contracts: Release as v3.0.0 and add reserved field to `ContractInfoOf`

### PR DESCRIPTION
**Pull request by athei**
Preparing pallet_contracts for release. It seems that [this PR](https://github.com/paritytech/substrate/pull/8104) requires the release of a `frame-benchmarks v3.1.0` before releasing contracts.

- Update README.md
- Add a CHANGELOG.md
- Bump version numbers and allow publishing.
- Add a `_reserved` field to `ContractInfoOf` for future proofing.

### CODE REVIEW

File: frame/contracts/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+The semantic versioning garantuees do cover the interface to the substrate runtime which
```
Comment (ascjones) [inline frame/contracts/CHANGELOG.md:8]:
```suggestion
The semantic versioning guarantees do cover the interface to the substrate runtime which
```

Comment (ascjones) [reply frame/contracts/CHANGELOG.md:8]: Also do you mean *do not*?

File: frame/contracts/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+includes this pallet as a dependency. This module will also add storage migrations whenever
+changes require this. Stability with regard to offchain tooling is explicitly excluded from
+this garantuee: For example adding a new field to an in-storage data structure will require
```
Comment (ascjones) [inline frame/contracts/CHANGELOG.md:11]:
```suggestion
this guarantee: For example adding a new field to an in-storage data structure will require
```

File: frame/contracts/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+changes to frontends to properly display it. However, those changes will still be regarded
+as a minor version bump.
+
+The interface provided to smart contracts will adhere to semver to with one exception: Even
+major version bumps will be backwards compatible with regard to already deployed contracts.
+In other words: Upgrading this pallet will not break pre-existing contracts.
+
+## [v3.0.0]
+
+This version constitutes the first release that brings any stability garantuess (see above).
```
Comment (ascjones) [inline frame/contracts/CHANGELOG.md:21]:
```suggestion
This version constitutes the first release that brings any stability guarantees (see above).
```

**Review by ascjones — APPROVED** [review]
LGTM % typos

File: frame/contracts/CHANGELOG.md (hunk 1 above)
Comment (athei) [reply frame/contracts/CHANGELOG.md:8]: No I think it is correct as I have written it down: It  is covered as the runtime declares pallet_contracts as dependency in its Cargo.toml. Not sure why you think I mean the opposite. Should that be expressed in a different way then?

Comment (ascjones) [reply frame/contracts/CHANGELOG.md]: No it was just because of the word `do` on its own, rather than the content of the sentence, so was just checking.

Comment (athei) [reply frame/contracts/CHANGELOG.md]: I removed the word.

**Comment by 2075** [comment]
kan i haz?

**Comment by athei** [comment]
> kan i haz?

You will. Just a bit more patience :)

**Comment by atenjin** [comment]
Hope this pr could be merged today...

**Review by gnunicorn — APPROVED** [review]

**Pull request merged**